src/index.js

```diff
@@ -505,7 +501,7 @@ const viewPropositionsCache = {};
  * `applyPersonalization`.
  */
 export async function getPersonalizationForView(viewName = '__view__') {
```

Comment:
`__view__` is a special scope in Target. My recommendation is to not set the `viewName: __view__`. 

Because by adding it by default to a `viewName` will lead into race conditions:
- at Activity **Authoring** time - the SPA view experience will be set to a page-wide scope
- at **Delivery** time Web SDK has a special treatment for this type of scope

Reply:
Web SDK recommendation is to fetch propositions at page-load and at every view change to call sendEvent:
```
sendEvent({
    renderDecisions: true,
    xdm: {
      web: {
        webPageDetails: { viewName: "home" },
      },
    });
```
When a React component gets re-rendered they should call:
 ```
applyPropositions({viewName: "home"});
```
In this case Web SDK will fetch from it's cache the propositions for that view and re-render them.

OR

Store the result.propositions from `sendEvent` and pass them to `applyPropositions({propositions});
More [here](https://experienceleague.adobe.com/en/docs/experience-platform/web-sdk/personalization/rendering-personalization-content#:~:text=renderAttempted%3A%20true%20flag.-,Use%20case%201%3A%20Re%2Drender%20single%2Dpage%20application%20view%20propositions,-The%20use%20case).